WooCommerce/Classes/POS/Controllers/PointOfSaleOrderController.swift

```diff
@@ -51,16 +61,16 @@ protocol PointOfSaleOrderControllerProtocol {
     private(set) var orderState: PointOfSaleInternalOrderState = .idle
     private var order: Order? = nil
 
-    @MainActor
+    @MainActor @discardableResult
     func syncOrder(for cartItems: [CartItem],
-                   retryHandler: @escaping () async -> Void) async {
+                   retryHandler: @escaping () async -> Void) async -> Result<SyncOrderState, Error> {
         let posCartItems = cartItems.map {
             POSCartItem(item: $0.item, quantity: Decimal($0.quantity))
         }
 
         guard !orderState.isSyncing,
               !posCartItems.matches(order: order) else {
-            return
+            return .failure(SyncOrderStateError.syncFailure)
```

Comment:
If `posCartItems.matches(order: order)` it means that that  `.success(.orderNotChanged)` not shat syncing failed.

Reply:
Appreciated, I got confused with the negative guard 🙇 fixed here and added a new case and test: https://github.com/woocommerce/woocommerce-ios/pull/15148/commits/af7ce8f28fa33304f031bf1553f9c0b2e403248c